httpserver/handlers.go

```diff
@@ -116,19 +117,19 @@ func (server *Server) verify(ctx context.Context, w http.ResponseWriter, r *http
 				}
 
 				if cacheProvider != nil {
-					logrus.Debugf("cache miss for subject %v", resolvedSubjectReference)
+					logger.GetLogger(ctx, server.LogOption).Debugf("cache miss for subject %v", resolvedSubjectReference)
 					if !cacheProvider.SetWithTTL(ctx, fmt.Sprintf(cache.CacheKeyVerifyHandler, resolvedSubjectReference), result, server.CacheTTL) {
-						logrus.Warnf("unable to insert cache entry for subject %v", resolvedSubjectReference)
+						logger.GetLogger(ctx, server.LogOption).Warnf("unable to insert cache entry for subject %v", resolvedSubjectReference)
 					}
 				}
 
 				if res, err := json.MarshalIndent(result, "", "  "); err == nil {
-					logrus.Infof("verify result for subject %s: %s", resolvedSubjectReference, string(res))
+					logger.GetLogger(ctx, server.LogOption).Infof("verify result for subject %s: %s", resolvedSubjectReference, string(res))
 				}
 			}
 
 			returnItem.Value = fromVerifyResult(result, server.GetExecutor().PolicyEnforcer.GetPolicyType(ctx))
-			logrus.Debugf("verification: execution time for image %s: %dms", resolvedSubjectReference, time.Since(routineStartTime).Milliseconds())
+			logger.GetLogger(ctx, server.LogOption).Debugf("verification: execution time for image %s: %dms", resolvedSubjectReference, time.Since(routineStartTime).Milliseconds())
```

Comment:
should we have a logger variable similar to https://github.com/deislabs/ratify/blob/80122da99115212a9d591cfca79ea4a72c486637/pkg/manager/manager.go#L66?

Reply:
it's a bit different from the `setupLog`. `setupLog` remains unchanged across requests. But `logger.GetLogger()` is constructing a logger from the parent context where we get different `trace-id`. Therefore, we put the static and shared values within `Option` which behaves similarly to `setupLog`.